code/go/pkg/validator/api.go

```
@@ -0,0 +1,177 @@
+
+// newFromZip is the internal constructor used by both NewFromZip (ModeBuild) and
+// the deprecated ValidateFromZip wrapper (ModeLegacy).
+func newFromZip(mode Mode, zipPath string, opts ...Option) (_ *Validator, err error) {
+	r, openErr := zip.OpenReader(zipPath)
+	if openErr != nil {
+		return nil, fmt.Errorf("failed to open zip file (%s): %w", zipPath, openErr)
+	}
+	// Close the reader on any error path; on success the Validator takes ownership.
+	defer func() {
+		if err != nil {
+			r.Close()
```

golangci-lint (2.12.2)

[error] Error return value of `r.Close` is not checked

(errcheck)